tests/produce_fetch.rs

```diff
@@ -0,0 +1,249 @@
+use std::{net::TcpStream, time::Duration};
+
+use bytes::{Bytes, BytesMut};
+use kafka_protocol::{
+    messages::{
+        create_topics_request::CreatableTopic,
+        fetch_request::{FetchPartition, FetchTopic},
+        produce_request::{PartitionProduceData, TopicProduceData},
+        ApiKey, CreateTopicsRequest, CreateTopicsResponse, FetchRequest, FetchResponse,
+        ProduceRequest, ProduceResponse, RequestHeader, TopicName,
+    },
+    protocol::{Builder, StrBytes},
+    records::{
+        Compression, Record, RecordBatchDecoder, RecordBatchEncoder, RecordEncodeOptions,
+        TimestampType,
+    },
+};
+
+mod common;
+use common::*;
+
+#[test]
+fn record_batch_produce_fetch() {
```

Comment:
Considering these two tests are variants of each other, maybe name this one as `message_set_v2_produce_fetch`?